streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1097,15 +1097,20 @@ long decodeTimestamp(final String encryptedString) {
         if (encryptedString.isEmpty()) {
             return RecordQueue.UNKNOWN;
         }
-        final ByteBuffer buffer = ByteBuffer.wrap(Base64.getDecoder().decode(encryptedString));
-        final byte version = buffer.get();
-        switch (version) {
-            case LATEST_MAGIC_BYTE:
-                return buffer.getLong();
-            default:
-                log.warn("Unsupported offset metadata version found. Supported version {}. Found version {}.",
-                         LATEST_MAGIC_BYTE, version);
-                return RecordQueue.UNKNOWN;
+        try {
+            final ByteBuffer buffer = ByteBuffer.wrap(Base64.getDecoder().decode(encryptedString));
+            final byte version = buffer.get();
+            switch (version) {
+                case LATEST_MAGIC_BYTE:
+                    return buffer.getLong();
+                default:
+                    log.warn("Unsupported offset metadata version found. Supported version {}. Found version {}.",
+                            LATEST_MAGIC_BYTE, version);
+                    return RecordQueue.UNKNOWN;
+            }
+        } catch (final IllegalArgumentException argumentException) {
```

Comment:
Sorry for the follow up comment (should have added this right away): why are we catching `IllegalArgumentException`? Would it be better to just catch `Exception` to cover all potential error cases? `buffer.get()` could throw a `BufferUnderflowException` for example?

Reply:
No problem, that's why reviews like this exists.
I didn't really think about the exceptions thrown by the ByteBuffer in this case

Reply:
@mjsax I've updated the implementation, can you take a look again?